### PR TITLE
Swagger with spec record (cljs support)

### PR DESCRIPTION
**Pull request by nenadalm**
Hi. I was playing with reitit and Macchiato (https://github.com/macchiato-framework/macchiato-core/issues/7) but I am not sure how to generate spec. I've tried to follow guide for ring (https://metosin.github.io/reitit/ring/swagger.html).

It looks like reitit is converting given spec in `:parameters` into spec records and then I get following error in cljs:
```
#object[Error Error: No protocol method IKVReduce.-kv-reduce defined for type : [object Object]]
```

it comes from this function: https://github.com/metosin/spec-tools/blob/cb4b44845ae379e2b0f0f2e145aa1570fe00513b/src/spec_tools/swagger/core.cljc#L136

It seems to sort of work for clj, the result is just a bit different with spec records in the test (added test is just copy-paste of the test above, just with converted spec to spec records). Spec record is missing `:title` key in `body.schema`.

### CODE REVIEW

**Comment by nenadalm** [comment]
It looks like the problem is in clojurescript, because records don't implement the protocol: https://dev.clojure.org/jira/browse/CLJS-1297

File: test/cljc/spec_tools/swagger/core_test.cljc [hunk 1]
```diff
@@ -239,7 +239,59 @@
...
+                          :name "spec-tools.swagger.core-test/id"
+                          :description ""
+                          :type "string"
+                          :required true}
+                         {:in "body",
+                          :name "spec-tools.swagger.core-test/address",
+                          :description "",
+                          :required true,
+                          :schema {:type "object",
+                                   ;; :title "spec-tools.swagger.core-test/address",
```
Comment (nenadalm) [inline test/cljc/spec_tools/swagger/core_test.cljc]: Is this intentional? If spec is used instead of spec records, the `:title` is present.

Comment (ikitommi) [reply test/cljc/spec_tools/swagger/core_test.cljc]: I think the title should be present in both.

**Comment by ikitommi** [comment]
Thanks, will fix the title and push out a release

**Pull request merged**